posthog/api/test/test_event.py

```diff
@@ -264,14 +265,23 @@ def test_before_and_after(self):
             self.assertEqual(response["results"][0]["id"], event2.pk)
 
         def test_pagination(self):
-            events = []
             person_factory(team=self.team, distinct_ids=["1"])
-            for index in range(0, 150):
-                event_factory(team=self.team, event="some event", distinct_id="1")
+            for _ in range(0, 150):
+                event_factory(
+                    team=self.team, event="some event", distinct_id="1", timestamp=datetime(2020, 1, 1, 12, 0, 0, _)
+                )
             response = self.client.get("/api/event/?distinct_id=1").json()
+            self.assertEqual(len(response["results"]), 100)
             self.assertIn("http://testserver/api/event/?distinct_id=1&before=", response["next"])
 
             page2 = self.client.get(response["next"]).json()
+            from posthog.ee import check_ee_enabled
+
+            if check_ee_enabled():
+                from ee.clickhouse.client import sync_execute
+
+                self.assertEqual(sync_execute("select count(*) from events")[0][0], 150)
+
             self.assertEqual(len(page2["results"]), 50)
```

Comment:
@EDsCODE @fuziontech I am ridiculously stuck on this. For some reason running this locally passed, but in github actions `len(page2["results"])` returns 99 instead of 50. I've tried a bunch of stuff to try and debug this (see above) but nothing makes sense to me. Any thoughts?

Reply:
Code that this test calls is [here](https://github.com/PostHog/posthog/blob/master/ee/clickhouse/views/events.py#L58)

Reply:
giving it a look right now